File: prog3/trab2/Genre.cpp
/**
 * Genre class implementation
 */

#include "Genre.hpp"

// Constructor
Genre::Genre(const string& initials, const string& name)
    : initials(initials), name(name)
    {}

// Functions
void Genre::addMedia(Media& media) {
    this->medias.push_back(&media);
}

int Genre::getMediaCount() {
    return this->medias.size();
}

// Getters
string& Genre::getInitials() { return this->initials; }
string& Genre::getName() { return this->name; }

// Setters
void Genre::setName(const string& s) { this->name = s; }
void Genre::setInitials(const string& s) { this->initials = s; }

// Destructor
Genre::~Genre() {}

File: prog3/trab1/src/Prog3_Trab1/Genre.java
package Prog3_Trab1;

import java.util.List;
import java.io.Serializable;
import java.util.LinkedList;

final class Genre implements Serializable {
	
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	private String initials;
	private String name;
	private List<Media> medias;
	
	public Genre(String initials) {
		this.initials = initials;
		this.medias = new LinkedList<Media>();
	}
	
	public String getInitials() {
		return initials;
	}
	
	public String getName() {
		return name;
	}
	
	public void setName(String name) {
		this.name = name;
	}
	
	public void addMedia(Media media) {
		this.medias.add(media);
	}
	
	public int getMediaCount() {
		return this.medias.size();
	}
	
	@Override
	public String toString() {
		return this.initials+";"+this.name;
	}
}